src/Components/Contentlet.js

```diff
@@ -1,60 +1,24 @@
-import React from 'react';
-import DotContentlet from '../libs/DotContentlet';
-import NewsWidget from './News/NewsWidget'
+import React, { Component } from 'react';
 
-function createMarkup(html) { return {__html: html}; };
+import Loadable from 'react-loadable';
 
-const ContentGeneric = ({data}) => {
-    return (
-        <div className="content">
-            <h2>{data.title}</h2>
-            <span dangerouslySetInnerHTML={createMarkup(data.body)} />
-        </div>
-    )
-}
-
-const CalendarEvent = ({data}) => {
-    return (
-        <div className="event">
-            <h2>{data.title}</h2>
-            <time>{data.startDate}</time>
-            <time>{data.startEndDate}</time>
-            <span dangerouslySetInnerHTML={createMarkup(data.description)} />
-        </div>
-    )
-}
-
-const SimpleWidget = ({data}) => {
-    return (
-        <div className="widget">
-            <h2>{data.widgetTitle}</h2>
-        </div>
-    )
-}
+import CantRender from './CantRender';
 
-const getComponent = (data) => {
-    switch (data.contentType) {
-        case 'webPageContent':
-            return ContentGeneric;
-        case 'calendarEvent':
-            return CalendarEvent;
-        case 'SimpleWidget':
-            return SimpleWidget;
-        case 'NewsWidgets':
-            return NewsWidget;
-        default: 
-            return ContentGeneric;
+const Loading = (props) => {
+    if (props.error)  {
+        return <CantRender color="warning" title="Error"><p>{props.error.message}</p></CantRender>
     }
-}
-
 
-const Contentlet = ({data}) => {
-    const Component = getComponent(data);
-    return (
-        <DotContentlet data={data}>
-            <Component data={data} />
-        </DotContentlet>
-    )
+    return <h1>Loading</h1>;
 };
 
-export default Contentlet;
+export default class Contentlet extends Component {
+    render() {
+        const Component = Loadable({
+            loader: () => import(`./DotCMS/${this.props.data.contentType}`),
+            loading: Loading
```

Comment:
If no component found... we render a Loading component and if there is an error we can show the error.